iceoryx_utils/test/moduletests/test_cxx_vector.cpp

```diff
@@ -1036,3 +1043,42 @@ TEST_F(vector_test, PartiallyEqualVectorsWithDifferentCapacityAreNotEqual)
     EXPECT_FALSE(a == b);
     EXPECT_TRUE(a != b);
 }
+
+TEST_F(vector_test, FullVectorDestroysElementsInReverseOrder)
+{
+    static constexpr uint64_t VECTOR_CAPACITY = 35U;
+    {
+        vector<CTorTest, VECTOR_CAPACITY> sut;
+
+        for (uint64_t i = 0U; i < VECTOR_CAPACITY; ++i)
+        {
+            sut.emplace_back(i);
+        }
+    }
+
+    ASSERT_THAT(dtorOrder.size(), Eq(VECTOR_CAPACITY));
+    for (uint64_t i = 0U; i < VECTOR_CAPACITY; ++i)
+    {
+        EXPECT_THAT(dtorOrder[i], Eq(VECTOR_CAPACITY - i - 1U));
+    }
+}
+
+TEST_F(vector_test, PartiallyFullVectorDestroysElementsInReverseOrder)
+{
+    static constexpr uint64_t VECTOR_CAPACITY = 40U;
+    static constexpr uint64_t VECTOR_SIZE = 20U;
+    {
+        vector<CTorTest, VECTOR_CAPACITY> sut;
+
+        for (uint64_t i = 0U; i < VECTOR_SIZE; ++i)
+        {
+            sut.emplace_back(i + VECTOR_CAPACITY);
+        }
+    }
+
+    ASSERT_THAT(dtorOrder.size(), Eq(VECTOR_SIZE));
+    for (uint64_t i = 0U; i < VECTOR_SIZE; ++i)
+    {
+        EXPECT_THAT(dtorOrder[i], Eq(VECTOR_SIZE - i - 1U + VECTOR_CAPACITY));
```

Comment:
Adding VECTOR_CAPACITY is only needed due to translation above, I do not see why it would be needed but if we use it it should be done with an independent constant for clarity.